### ISSUE TITLE
Can you offer C++ interface to this awesome library.

### ISSUE THREAD

**Issue by linjiapro** [facebookresearch/pytorch3d#371]
## 🚀 Feature

This library is mostly written in python and can integrate with the python version of PyTorch gracefully. Could you also offer a C++ version, since PyTorch has C++ version Libtorch?

## Motivation

C++ is an important language in computer vision, and many projects are already written in that language. Pytorch has a C++ version too (Libtorch). Python does offer C++ binding, but  that will be not enough for the following scenario: 

You have a neural network, and it is implemented in the C++ side. You want to integrate a pytorch3D network into that existing neural network to calculate some costs and backpropagate. I don't think C++ and Python binding can do the backpropagation. Therefore, a C++ version of this project is needed, in order to gracefully integrate this awesome library into an existing C++ project.

Huge thanks in advance!

**Comment by bottler** [comment]
PyTorch3D is unlikely to be fully implemented in C++ because we want a wide range of researchers to be able to quickly understand and hack on it. 

I think your scenario can be achieved by passing the data in the middle of the model to Python. I think this might be quite complicated but I don't think it will break the gradient calculation.

A more realistic way to call some pytorch3d from C++ is using torchscript. If there is a limited amount of pytorch3d functionality which you need to be made available, this might be quite achievable. What are you trying to do in particular?
